nexus/db-model/src/sled.rs

```diff
@@ -55,6 +60,11 @@ impl Sled {
             ip: ipv6::Ipv6Addr::from(addr.ip()),
             port: addr.port().into(),
             last_used_address,
+            // TODO: these fields should be populated
```

Comment:
Apologies if this is a dumb question, but I'm not sure from the context - at least `slot` of these has to come from MGS, but the `id`/`addr` definitely do not come from MGS. Is it reasonable to expect `Sled::new()` to already have combined info from both sled-agent and MGS?

Reply:
Possibly! I'm not really sure. Maybe a better question for @smklein? I'm running off the default assumption that the information from MGS is populated at a later time. If we had that initial information from when the sled record was created though, that would be nice.

Reply:
Well, right now we don't have any integration with MGS, so the naive answer for today is "we only have the self-reported information from the sled".

Admittedly, the sled *can*- and maybe *should* - self-report the part/serial/revision values as part of the call to nexus.

The handoff looks like the following:

- Sled agent boots
- Sled agent [sends a request to Nexus, indicating that it's alive](https://github.com/oxidecomputer/omicron/blob/5c0ac704e81ae79370ee743cb5168d7aeb949e6c/nexus/src/internal_api/http_entrypoints.rs#L66-L86)
- Nexus calls this function, stores the record for the sled in the DB.

In the future, we could make Nexus' implementation of this endpoint call MGS if we don't already know the slot number of the sled - that would give us an opportunity to ensure that "all sleds known to Nexus have a non-optional slot". **Personally, I would prefer if we required some of this information up-front, rather than hoping it gets back-filled later**. Trying to operate normally with sleds that have unknown identifiers and unknown versions seems like it'll just cause us pain.

(related: In the future, it might make sense to split Nexus' awareness of sleds in two. Currently, we only really know about sleds if the host has fully booted. Later, it probably makes sense to track "sleds that are visible to MGS" and "sleds which have fully booted their hosts" as separate entities)

Reply:
I think ultimately it will have to. As I understand it, without having the part/serial/revision we actually have no way of matching Nexus' notion of a sled to that MGS returns. If we know which cubby the sled is in we can make call MGS to get more information about it but because sleds can be physically moved that feels like a weak guarantee to me. I haven't wired it up that way in this PR because none of those mechanisms exist yet.

I know it's an aside, but I'm a little hesitant about having multiple notions of a sled. It seems to me that that would be much harder to reconcile from a product perspective and be a potential source of syncing bugs.